streaming/src/test/java/org/apache/spark/streaming/JavaAPISuite.java

```diff
@@ -1244,7 +1244,15 @@ public void testCountByValue() {
     JavaTestUtils.attachTestOutputStream(counted);
     List<List<Tuple2<String, Long>>> result = JavaTestUtils.runStreams(ssc, 3, 3);
 
-    Assert.assertEquals(expected, result);
+
+
+    for (int i=0;i<expected.size();i++) {
```

Comment:
This only asserts that the actual result is a superset of the expected results. It won't catch the case that it contains unexpected values. You could check both ways.